<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html>
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Plotting functions in GUESSFM • GUESSFM</title>
<!-- jquery --><script src="https://code.jquery.com/jquery-3.1.0.min.js" integrity="sha384-nrOSfDHtoPMzJHjVTdCopGqIqeYETSXhZDFyniQ8ZHcVy08QesyHcnOUpMpqnmWq" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://maxcdn.bootstrapcdn.com/bootswatch/3.3.7/simplex/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous">
<script src="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha384-Tc5IQib027qvyjSMfHjOMaLkfuWVxZxUPnCJA7l2mCWNIpG9mGCD8wGNIcPD7Txa" crossorigin="anonymous"></script><!-- Font Awesome icons --><link href="https://maxcdn.bootstrapcdn.com/font-awesome/4.6.3/css/font-awesome.min.css" rel="stylesheet" integrity="sha384-T8Gy5hrqNKT+hzMclPo118YTQO6cYprQmhrYwIiQ/3axmI1hQomh7Ud2hPOy8SP1" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/1.7.1/clipboard.min.js" integrity="sha384-cV+rhyOuRHc9Ub/91rihWcGmMmCXDeksTtCihMupQHSsi8GIIRDG0ThDc3HGQFJ3" crossorigin="anonymous"></script><!-- sticky kit --><script src="https://cdnjs.cloudflare.com/ajax/libs/sticky-kit/1.1.3/sticky-kit.min.js" integrity="sha256-c4Rlo1ZozqTPE2RLuvbusY3+SU1pQaJC0TjuhygMipw=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Plotting functions in GUESSFM">
<meta property="og:description" content="">
<meta name="twitter:card" content="summary">
<!-- mathjax --><script src="https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">GUESSFM</a>
        <span class="label label-default" data-toggle="tooltip" data-placement="bottom" title="Released package">1.0.1</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fa fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/groups.html">Grouping and tagging SNPs</a>
    </li>
    <li>
      <a href="../articles/plotting.html">Plotting functions in GUESSFM</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right"></ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      
      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1>Plotting functions in GUESSFM</h1>
                        <h4 class="author">Chris Wallace</h4>
            
            <h4 class="date"><code>�r Sys.Date()</code></h4>
      
      
      <div class="hidden name"><code>plotting.Rmd</code></div>

    </div>

    
    
<h1>
GUESSFM Plotting
</h1>
<p><font color="grey"> <em>Chris Wallace // <a href="http://www-gene.cimr.cam.ac.uk/staff/wallace">web</a> // <a href="mailto:chris.wallace%20at%20cimr.cam.ac.uk">email</a></em><br></font></p>
<div id="table-of-contents">
<h2 class="hasAnchor">
<a href="#table-of-contents" class="anchor"></a>
Table of Contents
</h2>
<div id="text-table-of-contents">
<ul>
<li>
<a href="#orgheadline1">1. Some example data</a>
</li>
<li>
<a href="#orgheadline2">2. Prior/posterior number of SNPs in the model</a>
</li>
<li>
<a href="#orgheadline3">3. SNP groups</a>
</li>
<li>
<a href="#orgheadline4">4. Patterns of SNPs selected</a>
</li>
</ul>
</div>
</div>
<div id="some-example-data" class="section level1">
<h1 class="hasAnchor">
<a href="#some-example-data" class="anchor"></a>Some example data</h1>
<p>Let’s start with the data from the main vignette (GUESSFM Introduction):</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">library</span>(GUESSFM)</code></pre></div>
<pre><code>## Warning: replacing previous import 'data.table::melt' by 'reshape::melt'
## when loading 'GUESSFM'</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">mydir &lt;-<span class="st">  </span><span class="kw">system.file</span>(<span class="st">"extdata"</span>,<span class="dt">package=</span><span class="st">"GUESSFM"</span>)
(<span class="kw">load</span>(<span class="kw">file.path</span>(mydir,<span class="st">"simdata.RData"</span>)))</code></pre></div>
<pre><code>## [1] "X"    "Y"    "tags" "d"    "dx"   "sm"   "sp"</code></pre>
</div>
<div id="priorposterior-number-of-snps-in-the-model" class="section level1">
<h1 class="hasAnchor">
<a href="#priorposterior-number-of-snps-in-the-model" class="anchor"></a>Prior/posterior number of SNPs in the model</h1>
<p>The prior for each model is dependent only on the number of SNPs it contains. It is a beta-binomial model, and you need specify only the mean and overdispersion (variance relative to binomial) of this distrubtion.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">n&lt;-<span class="dv">100</span> <span class="co"># 100 SNPs in region</span>
  x &lt;-<span class="st"> </span><span class="dv">1</span><span class="op">:</span><span class="dv">10</span> <span class="co"># consider prior for up to 10 causal SNPs</span>
  xbar &lt;-<span class="st"> </span><span class="dv">3</span> <span class="co"># expect around 3 causal</span>

  ## a binomial prior
  y &lt;-<span class="st"> </span><span class="kw"><a href="../reference/snpprior.html">snpprior</a></span>(x, n, xbar)
  <span class="kw">plot</span>(x, y, <span class="dt">type=</span><span class="st">"h"</span>)

  ## is equivalent to
  y1<span class="fl">.0</span> &lt;-<span class="st"> </span><span class="kw"><a href="../reference/snpprior.html">snpprior</a></span>(x, n, xbar, <span class="dt">overdispersion=</span><span class="fl">1.0</span>)
  <span class="kw">points</span>(x, y1<span class="fl">.0</span>, <span class="dt">col=</span><span class="st">"red"</span>)

  ##larger values of overdispersion change the distribution:
  y1<span class="fl">.1</span> &lt;-<span class="st"> </span><span class="kw"><a href="../reference/snpprior.html">snpprior</a></span>(x, n, xbar, <span class="dt">overdispersion=</span><span class="fl">1.1</span>)
  y1<span class="fl">.5</span> &lt;-<span class="st"> </span><span class="kw"><a href="../reference/snpprior.html">snpprior</a></span>(x, n, xbar, <span class="dt">overdispersion=</span><span class="fl">1.5</span>)
  y2<span class="fl">.0</span> &lt;-<span class="st"> </span><span class="kw"><a href="../reference/snpprior.html">snpprior</a></span>(x, n, xbar, <span class="dt">overdispersion=</span><span class="fl">2.0</span>)
  <span class="kw">points</span>(x, y1<span class="fl">.1</span>, <span class="dt">col=</span><span class="st">"orange"</span>)
  <span class="kw">points</span>(x, y1<span class="fl">.5</span>, <span class="dt">col=</span><span class="st">"pink"</span>)
  <span class="kw">points</span>(x, y2<span class="fl">.0</span>, <span class="dt">col=</span><span class="st">"green"</span>)</code></pre></div>
<p><img src="plotting_files/figure-html/unnamed-chunk-2-1.png" width="700"></p>
<p><code>pp.nsnp</code> calculates a posterior distribution for the number of causal SNPs in the region, and with an argument <code>plot=TRUE</code> will also plot the distribution. If you supply the prior expected number of causal SNPs, you can compare the posterior to the prior.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw"><a href="../reference/pp.nsnp.html">pp.nsnp</a></span>(d,<span class="dt">plot=</span><span class="ot">TRUE</span>)</code></pre></div>
<p><img src="plotting_files/figure-html/unnamed-chunk-3-1.png" width="700"></p>
<pre><code>## $trait
##            0            1            2            3            4 
## 6.352132e-46 2.209605e-10 6.593048e-01 2.883969e-01 4.923457e-02 
##            5            6            7 
## 3.049965e-03 1.367187e-05 4.553833e-08</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">pp &lt;-<span class="st"> </span><span class="kw"><a href="../reference/pp.nsnp.html">pp.nsnp</a></span>(d,<span class="dt">plot=</span><span class="ot">TRUE</span>,<span class="dt">expected=</span><span class="dv">3</span>)</code></pre></div>
<p><img src="plotting_files/figure-html/unnamed-chunk-3-2.png" width="700"></p>
<p>So that’s quite nice. We had a vague-ish prior, spreading the prior weight across 0-7 SNP models, with an expected number of causal variants set at 3. We simulated data with two causal variants, and the posterior is peaked on 2 SNPs, with some weight on 3-4 SNPs.</p>
<p>The second time, we stored an underlying <code>pp.nsnp</code> object in <code>pp</code>. We can check a couple of quality control measures on this. Our priors are left skewed and unimodal, and so should our posteriors be:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw"><a href="../reference/qc.html">qc</a></span>(pp)</code></pre></div>
<pre><code>##       trait skewness nmodes  flag
## trait trait 1.318454      1 FALSE</code></pre>
</div>
<div id="snp-groups" class="section level1">
<h1 class="hasAnchor">
<a href="#snp-groups" class="anchor"></a>SNP groups</h1>
<p>In the introduction vignette, we formed a <code>snppicker</code> object, <code>sp</code>. We can extract the SNP groups from this and use it to generate a useful summary object</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">groups &lt;-<span class="st"> </span><span class="kw">as</span>(sp,<span class="st">"groups"</span>)
groups</code></pre></div>
<pre><code>## groups object, containing 5 SNPs in 2 groups.</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">library</span>(snpStats)</code></pre></div>
<pre><code>## Loading required package: survival</code></pre>
<pre><code>## Loading required package: Matrix</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">data</span>(for.exercise, <span class="dt">package=</span><span class="st">"snpStats"</span>) <span class="co"># for SNP information</span>
summx &lt;-<span class="st"> </span><span class="kw"><a href="../reference/guess.summ.html">guess.summ</a></span>(sm,<span class="dt">groups=</span>groups,<span class="dt">snps=</span>snp.support,<span class="dt">position=</span><span class="st">"position"</span>)
summx &lt;-<span class="st"> </span><span class="kw"><a href="../reference/scalepos.html">scalepos</a></span>(summx,<span class="dt">position=</span><span class="st">"position"</span>)

<span class="kw">library</span>(ggplot2)
signals &lt;-<span class="st"> </span><span class="kw"><a href="../reference/signal.plot.html">signal.plot</a></span>(summx)
signals</code></pre></div>
<p><img src="plotting_files/figure-html/unnamed-chunk-5-1.png" width="700"></p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">snps &lt;-<span class="st"> </span><span class="kw"><a href="../reference/ggsnp.html">ggsnp</a></span>(summx)
snps</code></pre></div>
<p><img src="plotting_files/figure-html/unnamed-chunk-5-2.png" width="700"></p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">lds &lt;-<span class="st"> </span><span class="kw"><a href="../reference/ggld.html">ggld</a></span>(X, summx)
lds</code></pre></div>
<p><img src="plotting_files/figure-html/unnamed-chunk-5-3.png" width="700"></p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">chr &lt;-<span class="st"> </span><span class="kw"><a href="../reference/ggchr.html">ggchr</a></span>(summx)
chr</code></pre></div>
<p><img src="plotting_files/figure-html/unnamed-chunk-5-4.png" width="700"></p>
<p>Individually, perhaps only the signals plot is useful, which shows us two groups of SNPs which together have joint gMPPI close to 1. The red one contains four SNPs, and the blue group 1. The other tracks are useful to put this into some context. <code>snps</code> labels the SNPs, <code>lds</code> plots the r<sup>2</sup> between pairs of SNPs. Grouped SNPs are ordered so SNPs in the same group are together, but this is unlikley to reflect chromosome position. <code>chr</code> is a plot of linking the group position to the chromosome position.</p>
<p>If you have <code>ggbio</code> installed, then this function will line up the above plots</p>
<pre><code>library(ggbio)
tracks(chr,signals,snps,lds,heights=c(1,3,1,2))</code></pre>
<div class="figure">
<img src="summx-tracks.jpg" alt="img"><p class="caption">img</p>
</div>
<p>This sort of thing can be useful if you have other tracks (eg, functional information) you want to line up.</p>
</div>
<div id="patterns-of-snps-selected" class="section level1">
<h1 class="hasAnchor">
<a href="#patterns-of-snps-selected" class="anchor"></a>Patterns of SNPs selected</h1>
<p>It can be helpful to see which groups of SNPs are selected together in models, or never together. If groups of SNPs are never selected together, then they maybe alternative explanations for the phenotype but do not jointly explain for of the variance than either set alone. Sometimes, this means <code>snp.picker</code> has failed to group SNPs properly, and these groups should be combined. You can inspect whether this might be helpful using <code><a href="../reference/check.merge.html">check.merge()</a></code> and use <code><a href="../reference/groups.merge.html">groups.merge()</a></code> to do the merge.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw"><a href="../reference/pattern.plot.html">pattern.plot</a></span>(<span class="dt">SM=</span>sm,<span class="dt">groups=</span>groups)</code></pre></div>
<p><img src="plotting_files/figure-html/unnamed-chunk-6-1.png" width="700"></p>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="sidebar">
        <div id="tocnav">
      <h2 class="hasAnchor">
<a href="#tocnav" class="anchor"></a>Contents</h2>
      <ul class="nav nav-pills nav-stacked">
<li><a href="#some-example-data">Some example data</a></li>
      <li><a href="#priorposterior-number-of-snps-in-the-model">Prior/posterior number of SNPs in the model</a></li>
      <li><a href="#snp-groups">SNP groups</a></li>
      <li><a href="#patterns-of-snps-selected">Patterns of SNPs selected</a></li>
      </ul>
</div>
      </div>

</div>


      <footer><div class="copyright">
  <p>Developed by Chris Wallace.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="http://pkgdown.r-lib.org/">pkgdown</a>.</p>
</div>

      </footer>
</div>

  

  </body>
</html>
